tests/integration/agents/test_graph.py

```diff
@@ -0,0 +1,234 @@
+                    "`CrashLoopBackOff` state. The last termination reason was `StartError`"
+                    " with the message indicating a failure to create the containerd task "
+                    "due to a context cancellation."
+                ),
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content="why the pod is failing?"),
+            ],
+            "{'subtasks': [{'description': 'why the pod is failing?', 'assigned_to': 'KubernetesAgent'}]}",
+            False,
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content="what is the status of my cluster?"),
+            ],
+            '{"subtasks": [{"description": "what is the status of my cluster?", "assigned_to": "KubernetesAgent"}]}',
+            False,
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content="What is Kubernetes? Explain Kyma function"),
+            ],
+            '{"subtasks": [{"description": "What is Kubernetes?", "assigned_to": "KubernetesAgent"},'
+            '{"description": "Explain Kyma function", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app with python and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app with python", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+    ],
+)
+def test_planner(
```

Comment:
can you please add a comment that explains what the purpose of the test is, what is done etc.?